rasa/nlu/selectors/response_selector.py

```diff
@@ -490,97 +519,101 @@ def _resolve_intent_response_key(
-            # responses seem to be unavailable,
-            # likely an issue with the training data
-            # we'll use a fallback instead
-            rasa.shared.utils.io.raise_warning(
-                f"Unable to fetch responses for {label_intent_response_key} "
-                f"This means that there is likely an issue with the training data."
-                f"Please make sure you have added responses for this intent."
+            logger.debug(
+                f"Adding following selector key to message property: {selector_key}"
             )
-            label_responses = [{TEXT: label_intent_response_key}]
 
-        for label in label_ranking:
-            label[INTENT_RESPONSE_KEY] = (
-                self._resolve_intent_response_key(label) or label[INTENT_NAME_KEY]
+            utter_action_key = util.intent_response_key_to_template_key(
+                label_intent_response_key
             )
-            # Remove the "name" key since it is either the same as
-            # "intent_response_key" or it is the response text which
-            # is not needed in the ranking.
-            label.pop(INTENT_NAME_KEY)
-
-        selector_key = (
-            self.retrieval_intent
-            if self.retrieval_intent
-            else RESPONSE_SELECTOR_DEFAULT_INTENT
-        )
+            prediction_dict = {
+                RESPONSE_SELECTOR_PREDICTION_KEY: {
+                    "id": top_label["id"],
+                    RESPONSE_SELECTOR_RESPONSES_KEY: label_responses,
+                    PREDICTED_CONFIDENCE_KEY: top_label[PREDICTED_CONFIDENCE_KEY],
+                    INTENT_RESPONSE_KEY: label_intent_response_key,
+                    RESPONSE_SELECTOR_UTTER_ACTION_KEY: utter_action_key,
+                },
+                RESPONSE_SELECTOR_RANKING_KEY: label_ranking,
+            }
 
-        logger.debug(
-            f"Adding following selector key to message property: {selector_key}"
-        )
+            self._set_message_property(message, prediction_dict, selector_key)
 
-        # TODO: remove `RESPONSE_SELECTOR_RESPONSE_TEMPLATES_KEY` and
-        # `RESPONSE_SELECTOR_TEMPLATE_NAME_KEY` in Open Source 3.0.0
-        utter_action_key = util.intent_response_key_to_template_key(
-            label_intent_response_key
-        )
-        prediction_dict = {
-            RESPONSE_SELECTOR_PREDICTION_KEY: {
-                "id": top_label["id"],
-                RESPONSE_SELECTOR_RESPONSES_KEY: label_responses,
-                RESPONSE_SELECTOR_RESPONSE_TEMPLATES_KEY: label_responses,
-                PREDICTED_CONFIDENCE_KEY: top_label[PREDICTED_CONFIDENCE_KEY],
-                INTENT_RESPONSE_KEY: label_intent_response_key,
-                RESPONSE_SELECTOR_UTTER_ACTION_KEY: utter_action_key,
-                RESPONSE_SELECTOR_TEMPLATE_NAME_KEY: utter_action_key,
-            },
-            RESPONSE_SELECTOR_RANKING_KEY: label_ranking,
-        }
+            if out and DIAGNOSTIC_DATA in out:
```

Comment:
there is a flag for the diagnostic thing on `execution_context` which we should use instead

Reply:
Ah I see, then I should replace the current if statement with a `if self._execution_context.should_add_diagnostic_data`?
I also had a question - the default of this flag is `False`, how would it change to `True`?

Reply:
that's all manual in the first place so somebody has to specify that when making predictions (same as it's now probably)